src/main/java/seedu/address/model/event/exceptions/EventIsAlreadyLinkedToContactException.java

```diff
@@ -0,0 +1,23 @@
+package seedu.address.model.event.exceptions;
+
+import seedu.address.model.contact.Contact;
+
+/**
+ * Signals that the required association between contact and event already exists.
+ */
+public class EventIsAlreadyLinkedToContactException extends RuntimeException {
+    private final Contact linkedContact;
+
+    /**
+     * Constructs a new EventIsAlreadyLinkedToContactException for an existing linked contact
+     * of an event.
+     */
+    public EventIsAlreadyLinkedToContactException(Contact contact) {
+        super("The contact is already linked to the event");
+        this.linkedContact = contact;
+    }
+
+    public Contact getContact() {
```

Comment:
Thanks for clarifying! This is because I want to notify the user which contact is the exsiting associated contact after catch the exception in execute() method of the LinkCommand. However, the LinkCommand only keeps a list of associated contact where I am not able to tell which contact is exactly the one already linked. Thus, I am trying to get this information from the exceptioin caught. Do let me know if you have any suggestions on how to better do this please! Tks!

Reply:
Sounds good! I think this is a good way to retrieve the existing associated contact.